test/e2e/src/test-custom-types.js

```diff
@@ -144,6 +150,17 @@ test('u128', async t => {
   t.is((await t.context.client.u128({ u128: 1n })).result, 1n)
 })
 
+test('from', async (t) => {
+  // objects with different constructors will not pass deepEqual check
+  function constructorWorkaround(object) {
+    return JSON.parse(JSON.stringify(object));
+  }
+
+  const clientFromFrom = await clientForFromTest(t.context.contractId, t.context.publicKey, t.context.keypair);
```

Comment:
I think it might be worth moving these checks into their own file, rather than putting them in `test-custom-types.js`. Then you don't need to worry so much about having beefy helper functions in that file.

Reply:
Okay, I will consolidate that logic into its own file for just testing the `from` method and the Constructor.